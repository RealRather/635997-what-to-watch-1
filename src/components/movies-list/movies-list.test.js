import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list.jsx';

const mocks = [
  {
    id: `11`,
    name: `Fantastic`,
    src: `img/fantastic.jpg`
  },
  {
    id: `22`,
    name: `Bohemian`,
    src: `img/bohemian.jpg`
  },
  {
    id: `33`,
    name: `Macbeth`,
    src: `img/macbeth.jpg`
  },
  {
    id: `44`,
    name: `We need to`,
    src: `img/we-need-to.jpg`
  }];

it(`Render MoviesList`, () => {
  const movieNames = mocks;
  const clickHandler = jest.fn();
  const focusHandler = jest.fn();
  const tree = renderer
    .create(<MoviesList
      movieNames={movieNames}
      onClick={clickHandler}
      onFocus={focusHandler}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


