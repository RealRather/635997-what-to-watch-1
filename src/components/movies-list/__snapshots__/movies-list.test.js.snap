// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MoviesList 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onFocus={[MockFunction]}
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[MockFunction]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic"
        height="175"
        src="img/fantastic.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onFocus={[MockFunction]}
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[MockFunction]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian"
        height="175"
        src="img/bohemian.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onFocus={[MockFunction]}
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[MockFunction]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        src="img/macbeth.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onFocus={[MockFunction]}
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[MockFunction]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="We need to"
        height="175"
        src="img/we-need-to.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        We need to
      </a>
    </h3>
  </article>,
]
`;
